### DISCUSSION TITLE
vsgImGui context & Earth globe rotation

### DISCUSSION THREAD

**Discussion by ComradeMashkov** [pelicanmapping/rocky#46] [Q&A / Support] [answered]
Hey Glenn and Rocky community! First of all I'd like to say that Rocky did help a lot to me and my studying Vulkan and VulkanSceneGraph. However, now I'm working on improving my personal experience in the application management. Thus, I have a couple of integration questions about 2 things:

1. Is it possible to make ImGui context menu appear when right-clicking the application area of the Window? I managed to handle mouse buttons events but 1) it catches all button events even if I click on the GUI element; 2) I didn't manage to create a context menu even with the (1) issue. How should I properly do that?
2. I found [this](https://github.com/BrutPitt/imGuIZMO.quat) library that allows to rotate quaternions around 3d vector. Does Rocky Earth globe has some 4d vector for example that is rotated? Is it possible to rotate and move not the camera, but the Earth globe?

Thanks for reply!

**Comment by gwaldron** [comment, marked as answer]
In the latest Rocky we embedded the vsgimgui dependency since (a) it is very minimal, and (b) we wanted to customize some of the code anyway. You can take a look at that code (in the `rdemo` project) and see if you can figure out how to get the information you need. Otherwise I'd suggest asking the VSG folks for assistance.

You can put the `MapNode` under a transform and rotate it if you want, though the earth is in an ECEF reference frame and there might still be parts of Rocky that depend on that being the case. I find it is less confusing to simply keep the earth in absolute ECEF coordinate space.
